Kudu.Core/Jobs/BaseJobRunner.cs

```diff
@@ -72,17 +71,65 @@ protected string WorkingDirectory
 
         protected JobSettings JobSettings { get; set; }
 
-        private static int CalculateHashForJob(string jobBinariesPath)
+        internal static bool JobDirectoryHasChanged(
+            Dictionary<string, FileInfoBase> sourceDirectoryFileMap, 
+            Dictionary<string, FileInfoBase> workingDirectoryFileMap, 
+            Dictionary<string, FileInfoBase> cachedSourceDirectoryFileMap)
         {
-            var updateDatesString = new StringBuilder();
-            DirectoryInfoBase jobBinariesDirectory = FileSystemHelpers.DirectoryInfoFromDirectoryName(jobBinariesPath);
+            // enumerate all source directory files, and compare against the files
+            // in the working directory (i.e. the cached directory)
+            FileInfoBase foundEntry = null;
+            foreach (var entry in sourceDirectoryFileMap)
+            {  
+                if (workingDirectoryFileMap.TryGetValue(entry.Key, out foundEntry))
+                {
+                    if (entry.Value.LastWriteTimeUtc != foundEntry.LastWriteTimeUtc)
```

Comment:
I'm not sure this handles the exe.config file modifications we do now (for app settings). Previously when I had was using > for the comparison, the modifications made to exe.config wouldn't trigger a dirty directory. Now it does I believe. Might need to special case the exe.config file. Thoughts?